src/lib/Libattr/attr_node_func.c

```diff
@@ -1133,3 +1136,162 @@ node_ntype(attribute *new, void *pnode, int actmode)
 	}
 	return rc;
 }
+
+/**
+ *
+ * @brief
+ *
+ *	Returns the "external" form of the attribute 'val' given 'name'.
+ *
+ * @param[in] 	name - attribute name
+ * @param[in] 	val - attribute value
+ *
+ * @return char * - the external form for name=state: "3" -> "down,offline"
+ * @Note
+ *     	Returns a static value that can potentially get cleaned up on next call.
+ * 	Must use return value immediately!
+ */
+char *
+return_external_value(char *name, char *val)
+{
+	char *vns;
+
+
+	if ((name == NULL) || (val == NULL))
+		return ("");
+
+	if (strcmp(name, ATTR_NODE_state) == 0) {
+		return vnode_state_to_str(atoi(val));
+	} else if (strcmp(name, ATTR_NODE_Sharing) == 0) {
+		vns = vnode_sharing_to_str((enum vnode_sharing)atoi(val));
+		return (vns?vns:"");
+	} else if (strcmp(name, ATTR_NODE_ntype) == 0) {
+		return vnode_ntype_to_str(atoi(val));
+	} else {
+		return val;
+	}
+}
+
+/**
+ * @brief
+ *		Returns the "internal" form of the attribute 'val' given 'name'.
+ *
+ * @param[in]	name	-	attribute name
+ * @param[in]	val	-	attribute value
+ *
+ * @return char *	: the external form for name=state: "down,offline" -> "3"
+ * @Note
+ *     	Returns a static value that can potentially get cleaned up on next call.
+ * 		Must use return value immediately!
+ *
+ * @par MT-safe: No
+ */
+char *
+return_internal_value(char *name, char *val)
+{
+	static char ret_str[40];
+	enum vnode_sharing share;
+	int  v;
+
+	if ((name == NULL) || (val == NULL))
+		return ("");
+
+	if (strcmp(name, ATTR_NODE_state) == 0) {
+		v=str_to_vnode_state(val);
+		sprintf(ret_str, "%d", v);
+		return (ret_str);
+	} else if (strcmp(name, ATTR_NODE_Sharing) == 0) {
+		share = str_to_vnode_sharing(val);
+		if (share == VNS_UNSET)
+			return val;
+		sprintf(ret_str, "%d", share);
+		return (ret_str);
+	} else if (strcmp(name, ATTR_NODE_ntype) == 0) {
+		v = str_to_vnode_ntype(val);
+		if (v == -1)
+			return val;
+		sprintf(ret_str, "%d", v);
+		return (ret_str);
+	} else {
+		return val;
```

Comment:
If you want use parenthesis for return value here also use it for the sake of uniformity across the function.

Reply:
Done.